agent/lib/kontena/websocket_client.rb

```diff
@@ -168,6 +179,14 @@ def connect_client(ws)
       ws.disconnect # close socket
     end
 
+    def backoff_reconnect!
+      @backoff_interval += 2 # exponential would make backoff really long with only few attempts
```

Comment:
You generally want to have some maximum cap on the backoff interval... with a lot of clients, it also helps to randomize the actual backoff.

Reply:
good point, 60s max?

Reply:
Yes, definitely needs to be a max. 60s would seem reasonable to me. 

Also this might not be a concern but introducing some randomness might be a good idea. E.g. if there is a large number of nodes and they are all disconnected simultaneously it might be beneficial to stagger their reconnection attempts by throwing in some randomness.

Reply:
https://www.awsarchitectureblog.com/2015/03/backoff.html

https://github.com/kontena/kontena-ipam/blob/accff08e6033436aaee8738d078b1354662e7e64/lib/helpers.rb#L24

Reply:
Well there ya go.. :)  👍